librosa/display.py

```diff
@@ -435,7 +438,7 @@ def waveplot(y, sr=22050, max_points=5e4, x_axis='time', offset=0.0, max_sr=1000
         y_top = y
         y_bottom = -y
 
-    axes = plt.gca()
+    axes = ax if ax is not None else plt.gca()
```

Comment:
here you may wanna check to make sure users actually give an axis object (if not None), because often people will get confused when they use `plt.subplots` w/ more than one axis, it'll give then a numpy array and then the plotting calls will have stranger error messages.

(the way I usually do this is to have a hidden `_check_ax` function to make this check throughout, see https://github.com/librosa/librosa/pull/689/files#diff-1ad5980f89cf6076183eced46362487aR901 for example)